### PR TITLE
Filter developers by role level

### PR DESCRIPTION
**Pull request by themudassarhassan**
This PR adds the functionality to filter by `Role Level` and closes [this](https://github.com/joemasilotti/railsdevs.com/issues/294) issue.
### Pull request checklist

- [x] Your code contains tests relevant for code you modified
- [x] You have linted and tested the project with `bin/check`

### CODE REVIEW

File: app/components/developer_query_component.html.erb [hunk 1]
```diff
@@ -21,6 +21,34 @@
...
         <div class="hidden sm:flex sm:items-baseline sm:space-x-8">
+          <div data-controller="toggle" data-toggle-close-class="hidden" class="relative z-10 inline-block text-left">
+            <div>
+              <button type="button" data-action="toggle#toggle" class="group inline-flex items-center justify-center text-sm font-medium text-gray-700 hover:text-gray-900" aria-expanded="false">
+                <span><%= t(".role_level.title") %></span>
+                <%= inline_svg_tag "icons/solid/chevron_down.svg", class: "flex-shrink-0 -mr-1 ml-1 h-5 w-5 text-gray-400 group-hover:text-gray-500" %>
+              </button>
+            </div>
+
+            <div data-toggle-target="element" class="origin-top-right absolute right-0 w-64 mt-2 bg-white rounded-md shadow-2xl p-4 ring-1 ring-black ring-opacity-5 focus:outline-none hidden">
```
Comment (joemasilotti) [inline app/components/developer_query_component.html.erb:32]:
```suggestion
            <div data-toggle-target="element" class="origin-top-right absolute right-0 mt-2 bg-white rounded-md shadow-2xl p-4 ring-1 ring-black ring-opacity-5 focus:outline-none hidden">
```

**Review by joemasilotti — CHANGES_REQUESTED** [review]
This looks great! Three things:

1. Remove the fixed width (there's a review comment suggestion)
2. PR #304 was just merged which adds mobile filtering - can you apply your changes to mobile, too?
3. I'm going to hide this behind a feature flag for now - role level is still new and only a few folks have added it

**Comment by themudassarhassan** [comment]
I can tackle 1 and 2 myself, regarding point 3 I would need some information/guidance like do we have feature flag implemented now? and any code example would be helpful! Thanks

**Comment by joemasilotti** [comment]
For 3, I've been doing something similar to the following:

```ruby
# app/models/feature.rb

class Feature
  def self.enabled?(feature_name)
    case feature_name.to_sym
    when :role_level_filters
      !Rails.env.production?
    end
  end
end
```

```ruby
# some view

<% if Feature.enabled?(:role_level_filters) do %>
  <%= form.checkbox :role_level_junior %>
<% end %>
```

**Comment by joemasilotti** [comment]
`main` will also need to be merged in before you work on the mobile filtering.

**Comment by themudassarhassan** [comment]
Okay thanks! I'll look into it.

**Comment by themudassarhassan** [comment]
Hey @joemasilotti I have pushed the changes. Please have a look when you get some time. Thanks!

File: app/models/feature.rb [hunk 2]
```diff
@@ -3,6 +3,8 @@ def self.enabled?(feature_name)
     case feature_name.to_sym
     when :new_developer_fields_banner
       true
+    when :role_level_filters
+      true
```
Comment (joemasilotti) [inline app/models/feature.rb:7]:
```suggestion
      Rails.env.development?
```

**Review by joemasilotti — COMMENTED** [review]
Outside of the one request can you merge in `main`? Then we can merge this in!

**Comment by themudassarhassan** [comment]
Hey @joemasilotti I have merged main in and also this filtering feature was need to be enabled for `test` env for the tests to be passed. I have made that change as well.

**Comment by joemasilotti** [comment]
Looks great - thank you for the contribution!

**Pull request merged**